google-cloud-firestore/lib/google/cloud/firestore/document_listener.rb

```diff
@@ -64,7 +71,7 @@ def initialize doc_ref, &callback
         ##
         # @private
         def start
-          @listener.start
+          synchronize { @listener.start }
```

Comment:
We don't need these. The `Watch::Listener` calls already (should) synchronize their own methods. This class should be responsible for synchronizing on only its own data (which is basically just `@error_callbacks`).